src/bin/sage-cython

```diff
@@ -1,39 +1,26 @@
-#!/usr/bin/env sage-python
+#!/usr/bin/env python
 
-# This script is a deprecated wrapper around the "cython" program.
-# It is deprecated since Issue #27041 (Sage 8.7) because one should
+# This script is a wrapper around the "cython" program.
+# It was deprecated in Issue #27041 (Sage 8.7) because one should
 # simply use "cython" directly. We display deprecation messages whenever
 # "sage-cython" does something different from plain "cython".
 #
 # A stronger deprecation warning was added in #29923 (Sage 9.2).
+#
+# In Sage 10.3, the deprecated functionality is removed permanently.
+# However, the script is un-deprecated and re-purposed as a temporary workaround
+# for defects in Cython 3.0.x support for PEP 420 implicit namespace packages,
+# see https://github.com/sagemath/sage/pull/36228
```

Comment:
Please don't un-deprecate this. Instead it should be removed in favor of using an unpatched version of cython.

Right now, compiling cython files that use sage modules is broken because of namespace packages, and patching cython here and in sage_setup gives the misleading impression that it works.

This is the opposite of "fitting in the python/cython ecosystem".

Reply:
Note that it is a "temporary workaround".

Anyone report the namespace problem to Cython upstream yet?

Reply:
Temporary hack. And when/if cython fixes it we still have to ship the hack since distros are slow updating cython, etc.

Let's stick to stable features, not the bleeding edge.

Reply:
@tornaria Kindly limit the use of fighting words such as "hack".

> Let's stick to stable features, not the bleeding edge.

We have been successfully using Cython with namespace packages ever since 2021.

If anything, it is that our porting effort to Cython 3 is incomplete. 
- https://github.com/sagemath/sage/pull/36228 tracks a remaining task, that's all.